File: csharp/VS2008/netframework/Modules/10.API/22.Virtual Mode/SheetSelectorForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace VirtualMode
{
    public partial class SheetSelectorForm: Form
    {
        public SheetSelectorForm()
        {
            InitializeComponent();
        }

        public SheetSelectorForm(string[] SheetNames)
            : this()
        {
            foreach (string s in SheetNames)
            {
                SheetList.Items.Add(s);
            }

            SheetList.SelectedIndex = 0;
        }

        internal bool Execute()
        {
            return ShowDialog() == DialogResult.OK;
        }

        public string SelectedSheet
        {
            get
            {
                return Convert.ToString(SheetList.SelectedItem);
            }
        }

        public int SelectedSheetIndex
        {
            get
            {
                return SheetList.SelectedIndex;
            }
        }

        private void SheetList_DoubleClick(object sender, EventArgs e)
        {
            DialogResult = DialogResult.OK;
        }
    }
}

File: vb/VS2013/netframework/Modules/10.API/22.Virtual Mode/SheetSelectorForm.vb
Imports System.ComponentModel
Imports System.Text

Namespace VirtualMode
	Partial Public Class SheetSelectorForm
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Public Sub New(ByVal SheetNames() As String)
			Me.New()
			For Each s As String In SheetNames
				SheetList.Items.Add(s)
			Next s

			SheetList.SelectedIndex = 0
		End Sub

		Friend Function Execute() As Boolean
			Return ShowDialog() = System.Windows.Forms.DialogResult.OK
		End Function

		Public ReadOnly Property SelectedSheet() As String
			Get
				Return Convert.ToString(SheetList.SelectedItem)
			End Get
		End Property

		Public ReadOnly Property SelectedSheetIndex() As Integer
			Get
				Return SheetList.SelectedIndex
			End Get
		End Property

		Private Sub SheetList_DoubleClick(ByVal sender As Object, ByVal e As EventArgs) Handles SheetList.DoubleClick
			DialogResult = System.Windows.Forms.DialogResult.OK
		End Sub
	End Class
End Namespace
